# Use

The plan is to integrate the cassette so that the HIS3 marker is removed. We chose HIS3 since this is the location of the del construct in RS112 and derivatives. 

We should use a leu2 HIS3 background for the integration.

The resulting strain should be G418R his3 LEU2 and HygS. After recombination, the cells will be G418S his3 eu2 and HygR. The cassette will be amplified so that the 2µ and URA3 sequences are left out.






In [11]:
from pygenome import sg

In [12]:
from pydna.all import *

In [13]:
HIS3_locus = Dseqrecord( sg.stdgene["HIS3"].locus() )

The HIS3_locus sequence contain the whole HIS3 locus including promoter and terminator.

In [14]:
HIS3_locus.write("HIS3_locus.gb")

HIS3_locus.gb

Paulo suggested these primers for integration of cassette in the HIS3 locus

In [15]:
intprim1, intprim2 = parse('''
>F
CTT TCC CGC AAT TTT CTT TTT CTA TTA CTC TTG GCC TCC T    aaaactgtattataagta

>R
TAT ATA TAT CGT ATG CTG CAG CTT TAA ATA ATC GGT GTC A gcg TT AGT ATC GAA TCG ACA G
''')

After discussion we arrived at the final primers below:

In [16]:
intprim1, intprim2 = parse('''
>pPS1_his_f
CTT TCC CGC AAT TTT CTT TTT CTA TTA CTC TTG GCC TCC T    agagcttcaatttaattatatcagttattatcc

>pPS1_his_r
TAT ATA TAT CGT ATG CTG CAG CTT TAA ATA ATC GGT GTC A gcg TT AGT ATC GAA TCG ACA G
''')

In [17]:
print(intprim1.format("tab"))
print(intprim2.format("tab"))

pPS1_his_f	CTTTCCCGCAATTTTCTTTTTCTATTACTCTTGGCCTCCTagagcttcaatttaattatatcagttattatcc
pPS1_his_r	TATATATATCGTATGCTGCAGCTTTAAATAATCGGTGTCAgcgTTAGTATCGAATCGACAG


In [18]:
intprim1

Dseqrecord(-73)

In [19]:
intprim2

Dseqrecord(-61)

In [20]:
prd = pcr(intprim1, intprim2, pPS1)

NameError: name 'pPS1' is not defined

In [21]:
prd.figure()

NameError: name 'prd' is not defined

In [22]:
prd.program()

NameError: name 'prd' is not defined

In [70]:
prd.dbd_program()


Pfu-Sso7d (rate 15s/kb)                 |8049bp
Three-step|          30 cycles   |      |Tm formula: Pydna tmbresluc
98.0°C    |98.0°C                |      |SaltC 50mM
__________|_____          72.0°C |72.0°C|Primer1C 1.0µM
00min30s  |10s  \ 66.0°C ________|______|Primer2C 1.0µM
          |      \______/ 2min 0s|10min |GC 46%
          |        10s           |      |4-12°C


The prd variable below contains the 8049 bp PCR product.

limit for [phusion](https://www.neb.com/protocols/1/01/01/pcr-protocol-m0530) polymerase seems to be 10 kb.

The PCR product sequence can be downloaded below.

In [71]:
prd.write("prd.gb")

prd.gb

The integration of the cassette is simulated below:

In [72]:
asm = Assembly((his3,prd), max_nodes=3)

In [73]:
asm

Assembly:
Sequences........................: [2663] [8049]
Sequences with shared homologies.: [2663] [8049]
Homology limit (bp)..............: 25
Number of overlaps...............: 2
Nodes in graph(incl. 5' & 3')....: 4
Only terminal overlaps...........: No
Circular products................: 
Linear products..................: [9781] [9009] [8821] [1891] [1703] [931]

In [74]:
cassette_integrated_in_HIS3_locus = asm.assemble_linear()[0]

In [75]:
cassette_integrated_in_HIS3_locus.figure()

BK006948|40
         \/
         /\
         40|8049bp_PCR_prod|40
                            \/
                            /\
                            40|BK006948

In [76]:
cassette_integrated_in_HIS3_locus.write("cassette_integrated_in_HIS3_locus.gb")

cassette_integrated_in_HIS3_locus.gb

http://webpcr.appspot.com/

Screening primers for colony PCR:

    >A-HIS3
    TGACGACTTTTTCTTAATTCTCGTT 

    >D-HIS3
    GCTCAGTTCAGCCATAATATGAAAT 


In [1]:
from pydna.myprimers import list_primers

In [4]:
p1= list_primers[1044]

In [5]:
p2 = list_primers[1043]

In [7]:
p1.seq

Seq('TGACGACTTTTTCTTAATTCTCGTT', IUPACAmbiguousDNA())

In [9]:
p2.seq

Seq('GCTCAGTTCAGCCATAATATGAAAT', IUPACAmbiguousDNA())

In [10]:
his3 = Dseqrecord( sg.gene["HIS3"].locus() )

NameError: name 'Dseqrecord' is not defined

In [ ]:
PROD = pcr( primer_list[1044], primer_list[1043], his3)

In [ ]:
PROD.figure()

    >19_D-DFR1
    GACTCAGACAGGTTGAAAAGAAGAC

    >18_A-DFR1
    CAAAGGTTTGGTTTTCAGTTAAGAA